### PR TITLE
fixes minor bugs in eos_eapi module

### LINKED ISSUE #21908: eos_eapi: http_port not set/read correctly
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
eos_eapi

##### ANSIBLE VERSION
```
ansible 2.3.0 (test_ios_again 1dde9971ae) last updated 2017/02/23 09:43:19 (GMT +100)

```

##### SUMMARY
```yaml
- name: Change endpoint ports
  eos_eapi:
    enable_http: false
    http_port: 81
    https_port: 4443
    enable_local_http: yes 
    local_http_port: 8181
    provider: "{{ cli }}"
  register: eos_eapi_output
  connection: local


```
isn't setting ` http_port` to 81

##### STEPS TO REPRODUCE
 time ANSIBLE_ROLES_PATH=targets  ansible-playbook -vvvvv  -i  inventory.network-jobarker eos.yaml  -e "limit_to=eos_eapi" 

```yaml

TASK [eos_eapi : Change endpoint ports] **********************************************************************************************************************************************************************************************************
task path: /home/johnb/git/ansible-inc/ansible/test/integration/targets/eos_eapi/tests/cli/configure.yaml:3
<veos01> connection transport is cli
<veos01> socket_path: /home/johnb/.ansible/pc/ca5960d27a
Using module file /home/johnb/git/ansible-inc/ansible/lib/ansible/modules/network/eos/eos_eapi.py
<veos01> ESTABLISH LOCAL CONNECTION FOR USER: johnb
<veos01> EXEC /bin/sh -c 'echo ~ && sleep 0'
<veos01> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/johnb/.ansible/tmp/ansible-tmp-1487952531.0-181989874477958 `" && echo ansible-tmp-1487952531.0-181989874477958="` echo /home/johnb/.ansible/tmp/ansible-tmp-1487952531.0-181989874477958 `" ) && sleep 0'
<veos01> PUT /tmp/tmpod07C4 TO /home/johnb/.ansible/tmp/ansible-tmp-1487952531.0-181989874477958/eos_eapi.py
<veos01> EXEC /bin/sh -c 'chmod u+x /home/johnb/.ansible/tmp/ansible-tmp-1487952531.0-181989874477958/ /home/johnb/.ansible/tmp/ansible-tmp-1487952531.0-181989874477958/eos_eapi.py && sleep 0'
<veos01> EXEC /bin/sh -c 'python /home/johnb/.ansible/tmp/ansible-tmp-1487952531.0-181989874477958/eos_eapi.py; rm -rf "/home/johnb/.ansible/tmp/ansible-tmp-1487952531.0-181989874477958/" > /dev/null 2>&1 && sleep 0'
changed: [veos01] => {
    "ansible_facts": {
        "eos_eapi_urls": {
            "Management1": [
                "https://172.26.4.24:4443"
            ]
        }
    }, 
    "changed": true, 
    "commands": [
        "management api http-commands", 
        "no protocol http", 
        "protocol https port 4443", 
        "protocol http localhost port 8181", 
        "no shutdown"
    ], 
    "invocation": {
        "module_args": {
            "auth_pass": null, 
            "authorize": true, 
            "enable_http": false, 
            "enable_local_http": true, 
            "host": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "http": false, 
            "http_port": 81, 
            "https": null, 
            "https_port": 4443, 
            "local_http": true, 
            "local_http_port": 8181, 
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "port": null, 
            "provider": {
                "auth_pass": null, 
                "authorize": true, 
                "host": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
                "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
                "port": null, 
                "ssh_keyfile": null, 
                "timeout": null, 
                "transport": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
                "use_ssl": null, 
                "username": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
                "validate_certs": null
            }, 
            "socket": null, 
            "ssh_keyfile": null, 
            "state": "started", 
            "timeout": null, 
            "transport": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "use_ssl": null, 
            "username": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "validate_certs": null, 
            "vrf": "default"
        }
    }, 
    "session_name": "ansible_1487952531"
}

TASK [eos_eapi : eos_command] ********************************************************************************************************************************************************************************************************************
task path: /home/johnb/git/ansible-inc/ansible/test/integration/targets/eos_eapi/tests/cli/configure.yaml:14
<veos01> connection transport is cli
<veos01> socket_path: /home/johnb/.ansible/pc/ca5960d27a
Using module file /home/johnb/git/ansible-inc/ansible/lib/ansible/modules/network/eos/eos_command.py
<veos01> ESTABLISH LOCAL CONNECTION FOR USER: johnb
<veos01> EXEC /bin/sh -c 'echo ~ && sleep 0'
<veos01> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/johnb/.ansible/tmp/ansible-tmp-1487952535.46-243845812890914 `" && echo ansible-tmp-1487952535.46-243845812890914="` echo /home/johnb/.ansible/tmp/ansible-tmp-1487952535.46-243845812890914 `" ) && sleep 0'
<veos01> PUT /tmp/tmp0qer_B TO /home/johnb/.ansible/tmp/ansible-tmp-1487952535.46-243845812890914/eos_command.py
<veos01> EXEC /bin/sh -c 'chmod u+x /home/johnb/.ansible/tmp/ansible-tmp-1487952535.46-243845812890914/ /home/johnb/.ansible/tmp/ansible-tmp-1487952535.46-243845812890914/eos_command.py && sleep 0'
<veos01> EXEC /bin/sh -c 'python /home/johnb/.ansible/tmp/ansible-tmp-1487952535.46-243845812890914/eos_command.py; rm -rf "/home/johnb/.ansible/tmp/ansible-tmp-1487952535.46-243845812890914/" > /dev/null 2>&1 && sleep 0'
ok: [veos01] => {
    "changed": false, 
    "invocation": {
        "module_args": {
            "auth_pass": null, 
            "authorize": true, 
            "commands": [
                "show management api http-commands | json"
            ], 
            "host": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "interval": 1, 
            "match": "all", 
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "port": null, 
            "provider": {
                "auth_pass": null, 
                "authorize": true, 
                "host": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
                "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
                "port": null, 
                "ssh_keyfile": null, 
                "timeout": null, 
                "transport": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
                "use_ssl": null, 
                "username": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
                "validate_certs": null
            }, 
            "retries": 10, 
            "ssh_keyfile": null, 
            "timeout": null, 
            "transport": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "use_ssl": null, 
            "username": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "validate_certs": null, 
            "wait_for": null
        }
    }, 
    "stdout": [
        {
            "bytesIn": 4709220, 
            "bytesOut": 27832846, 
            "commandCount": 86844, 
            "corsOrigins": [], 
            "dscpValue": 0, 
            "enabled": true, 
            "executionTime": 9497.167350313473, 
            "hitCount": 25393, 
            "httpServer": {
                "configured": false, 
                "port": 80, 
                "running": false
            }, 
            "httpsServer": {
                "configured": true, 
                "port": 4443, 
                "running": false
            }, 
            "lastHitTime": 1487951691.6624587, 
            "localHttpServer": {
                "configured": true, 
                "port": 8181, 
                "running": false
            }, 
            "requestCount": 25162, 
            "unixSocketServer": {
                "configured": true, 
                "running": false
            }, 
            "urls": [
                "Management1 : https://172.26.4.24:4443"
            ], 
            "users": {
                "admin": {
                    "bytesIn": 4709220, 
                    "bytesOut": 27832846, 
                    "lastHitTime": 1487951691.662783, 
                    "requestCount": 25162
                }
            }, 
            "vrf": "default"
        }
    ], 
    "stdout_lines": [
        {
            "bytesIn": 4709220, 
            "bytesOut": 27832846, 
            "commandCount": 86844, 
            "corsOrigins": [], 
            "dscpValue": 0, 
            "enabled": true, 
            "executionTime": 9497.167350313473, 
            "hitCount": 25393, 
            "httpServer": {
                "configured": false, 
                "port": 80, 
                "running": false
            }, 
            "httpsServer": {
                "configured": true, 
                "port": 4443, 
                "running": false
            }, 
            "lastHitTime": 1487951691.6624587, 
            "localHttpServer": {
                "configured": true, 
                "port": 8181, 
                "running": false
            }, 
            "requestCount": 25162, 
            "unixSocketServer": {
                "configured": true, 
                "running": false
            }, 
            "urls": [
                "Management1 : https://172.26.4.24:4443"
            ], 
            "users": {
                "admin": {
                    "bytesIn": 4709220, 
                    "bytesOut": 27832846, 
                    "lastHitTime": 1487951691.662783, 
                    "requestCount": 25162
                }
            }, 
            "vrf": "default"
        }
    ]
}

TASK [eos_eapi : Expect endpoint ports to be set] ************************************************************************************************************************************************************************************************
task path: /home/johnb/git/ansible-inc/ansible/test/integration/targets/eos_eapi/tests/cli/configure.yaml:21
fatal: [veos01]: FAILED! => {
    "assertion": "http_config.stdout[0].httpServer.port == 81", 
    "changed": false, 
    "evaluated_to": false, 
    "failed": true
}

```

##### ACTUAL RESULTS
```yaml

```

### PR DESCRIPTION
**Pull request by privateip**
* warns on config argument no longer being needed
* lots of integration test cases updated

fixes #21908

### CODE REVIEW

File: test/integration/targets/eos_eapi/tests/cli/configure.yaml [hunk 1]
```diff
@@ -21,13 +21,13 @@
 - name: Expect endpoint ports to be set
   assert:
     that:
-        - http_config.stdout[0].httpServer.port == 81
+        - http_config.stdout[0].httpServer.port == 80
```
Comment (gundalow) [inline test/integration/targets/eos_eapi/tests/cli/configure.yaml:24]: On line 6 we set the http port to`81`, though we don't enable http
Is the "feature" that you can't set the port without enabling the service, therefore we need to check that the port is set to whatever it was when the service was last enabled?

File: test/integration/targets/eos_eapi/tests/cli/off.yaml [hunk 2]
```diff
@@ -10,6 +10,7 @@
   register: eos_eapi_output
   connection: local
 
+- pause: seconds=20
```
Comment (gundalow) [inline test/integration/targets/eos_eapi/tests/cli/off.yaml]: Could you please add a note to say why this is needed. 

I wonder if we night need to document this.

File: test/integration/targets/eos_eapi/tests/cli/zzz_reset.1 [hunk 3]
```diff
@@ -1,8 +1,9 @@
 - debug: msg="START CLI/RESET.YAML"
```
Comment (gundalow) [inline test/integration/targets/eos_eapi/tests/cli/zzz_reset.1:1]: This test will not get run anymore as it doesn't end in `.yaml`
If it's meant to be a note to return an EOS device to original configuration please add a comment saying that.

**Review by gundalow — CHANGES_REQUESTED** [review]

File: test/integration/targets/eos_eapi/tests/cli/off.yaml (hunk 2 above)
Comment (privateip) [reply test/integration/targets/eos_eapi/tests/cli/off.yaml]: when the service is disabled via command line, the system will gracefully shut it down... with the new connection plugin we iterate to quickly to the next task before the system has had a chance to disable the service so we need to briefly pause.   In retro spec, we should probably check for this condition in the module code and not just assume the service fully transitioned to stopped.

File: test/integration/targets/eos_eapi/tests/cli/configure.yaml (hunk 1 above)
Comment (privateip) [reply test/integration/targets/eos_eapi/tests/cli/configure.yaml:24]: Yes, agreed we should provide a warning that the port will not be changed due to the protocol not being enabled.

File: test/integration/targets/eos_eapi/tests/cli/zzz_reset.1 (hunk 3 above)
Comment (privateip) [reply test/integration/targets/eos_eapi/tests/cli/zzz_reset.1:1]: Lets chat on IRC about this test and badtransport.1.  I dont think the test cases are valid tests but would like to discuss before simply removing them.  To answer your question, yes, changing from `.yaml` to `.<anything else>` will stop it from running

**Pull request merged**